cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -396,13 +354,19 @@ public void upgrade(boolean continueOnFailure) throws Exception {
           continue;
         }
 
+        // if this was etlbatch or etlrealtime template
+        if ("ETLBatch".equals(appSpec.getName()) || "ETLRealtime".equals(appSpec.getName())) {
```

Comment:
yeah... there isn't really any other data to go on since templates were implemented using apps.